src/parquet/column/reader.h

```diff
@@ -44,6 +44,9 @@ class PARQUET_EXPORT ColumnReader {
   static std::shared_ptr<ColumnReader> Make(const ColumnDescriptor*,
       std::unique_ptr<PageReader>, MemoryAllocator* allocator = default_allocator());
 
+  virtual int64_t ReadBatchValues(int32_t batch_size, int16_t* def_levels,
+      int16_t* rep_levels, uint8_t* values, int64_t* values_read) = 0;
```

Comment:
I think it would be preferable to have `reinterpret_cast<T*>(values_buffer)` on the user end, to be more explicit. There is the `T` member of the typed column readers that you can use.